### PR TITLE
Allow sending request with different methods during connection

### PR DESCRIPTION
**Pull request by paulefoe**
I have a case where I need to send a POST request to connect, which I unable to do right now. This PR fixes it without breaking backward compatibility.

### CODE REVIEW

**Comment by paulefoe** [comment]
Kindly asking for a review from @rtfol

File: aiohttp_sse_client/client.py [hunk 1]
```diff
@@ -64,6 +64,7 @@ def __init__(self, url: str,
                  on_open=None,
                  on_message=None,
                  on_error=None,
+                 http_method='GET',
```
Comment (awarecan) [inline aiohttp_sse_client/client.py:67]: Can we move this into `option` parameter and also update the doc in the method comment? Please use `method` as the key name

**Review by awarecan — CHANGES_REQUESTED** [review]
Thanks for the contribution ❤️ 

Can you update the document for the new parameter, and write an unit test for it?

File: tests/web_platform_tests/test_request.py [hunk 2]
```diff
@@ -93,3 +96,25 @@ def on_message():
...
+    assert source._method == 'POST'
+    source._session.request = AsyncMock(
+        return_value=Mock(
+            status=200,
+            content_type=CONTENT_TYPE_EVENT_STREAM,
+            method='POST'
+        )
+    )
+    await source.connect()
+    assert source._response.method == 'POST'
```
Comment (paulefoe) [inline tests/web_platform_tests/test_request.py:120]: I'm wondering if this is useful at all?

With regards to the tests in general, for me every test was skipped unless I added the `pytest.mark.asyncio` flag, did I miss something?

File: tests/web_platform_tests/test_request.py [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
+from aiohttp_sse_client.client import CONTENT_TYPE_EVENT_STREAM
 
 from .const import WPT_SERVER
+from conftest import AsyncMock
+
 
 async def test_rquest_accept():
     """Test EventSource: Accept header.
-    
+
```
Comment (paulefoe) [inline tests/web_platform_tests/test_request.py:16]: I can revert this, pycharm did it automatiaclly

**Comment by paulefoe** [comment]
Also, thank you very much for the quick review! Awesome library :sunglasses:

File: tests/web_platform_tests/test_request.py [hunk 4]
```diff
@@ -93,3 +96,25 @@ def on_message():
     await test(404)
     await test(410)
     await test(503)
+
+
+@pytest.mark.asyncio
```
Comment (awarecan) [inline tests/web_platform_tests/test_request.py:101]: it should be unnecessary

File: tests/web_platform_tests/test_request.py [hunk 5]
```diff
@@ -93,3 +96,25 @@ def on_message():
...
+async def test_request_post_to_connect():
+    """Test EventSource option method for connection.
+
+    """
+    source = sse_client.EventSource(
+        WPT_SERVER + 'resources/cache-control.event_stream?pipe=sub',
```
Comment (awarecan) [inline tests/web_platform_tests/test_request.py:107]: this endpoint supports POST method
```suggestion
        WPT_SERVER + 'resources/message.py',
```

File: tests/web_platform_tests/test_request.py [hunk 6]
```diff
@@ -93,3 +96,25 @@ def on_message():
...
+    )
+
+    assert source._method == 'POST'
+    source._session.request = AsyncMock(
+        return_value=Mock(
+            status=200,
+            content_type=CONTENT_TYPE_EVENT_STREAM,
+            method='POST'
+        )
+    )
```
Comment (awarecan) [inline tests/web_platform_tests/test_request.py:118]: 'resources/message.py' supports POST method, we don't need to mock it

File: tests/web_platform_tests/test_request.py (hunk 2 above)
Comment (awarecan) [inline tests/web_platform_tests/test_request.py:120]:
```suggestion
    async for e in source:
        assert e.data == "data"
        break
    await source.close()

```

Comment (awarecan) [reply tests/web_platform_tests/test_request.py:120]:
```
> py.test
```
Simply execute py.test works fine for me on my MacOS with python 3.7.1

**Review by awarecan — CHANGES_REQUESTED** [review]

File: tests/web_platform_tests/test_request.py (hunk 2 above)
Comment (paulefoe) [reply tests/web_platform_tests/test_request.py]: I'm not going to lie, this is doesn't work for me on Linux with python 3.9.

All the test suites are skipped for me

Comment (paulefoe) [reply tests/web_platform_tests/test_request.py]: Same for 3.7.10

**Comment by paulefoe** [comment]
Thanks for taking care about tests! Anything else has to be done?

**Review by awarecan — APPROVED** [review]
Looks good now. Thanks for your contribution 🎉

**Comment by paulefoe** [comment]
Thanks for quick release! Keep up the good work :slightly_smiling_face:

**Pull request merged**